contracts/0.8.9/StakingRouter.sol

```diff
@@ -964,34 +964,37 @@ contract StakingRouter is AccessControlEnumerable, BeaconChainDepositor, Version
     ) external payable validStakingModuleId(_stakingModuleId) {
         if (msg.sender != LIDO_POSITION.getStorageAddress()) revert AppAuthLidoFailed();
 
+        StakingModule storage stakingModule = _getStakingModuleById(_stakingModuleId);
         uint256 depositsValue = msg.value;
-        if (depositsValue == 0 || depositsValue != _depositsCount * DEPOSIT_SIZE) {
-            revert InvalidDepositsValue(depositsValue);
-        }
 
-        bytes32 withdrawalCredentials = getWithdrawalCredentials();
-        if (withdrawalCredentials == 0) revert EmptyWithdrawalsCredentials();
+        if (depositsValue > 0) {
+            if (depositsValue != _depositsCount * DEPOSIT_SIZE) {
+                revert InvalidDepositsValue(depositsValue, _depositsCount);
+            }
 
-        StakingModule storage stakingModule = _getStakingModuleById(_stakingModuleId);
-        if (StakingModuleStatus(stakingModule.status) != StakingModuleStatus.Active) {
-            revert StakingModuleNotActive();
-        }
+            bytes32 withdrawalCredentials = getWithdrawalCredentials();
+            if (withdrawalCredentials == 0) revert EmptyWithdrawalsCredentials();
 
-        (bytes memory publicKeysBatch, bytes memory signaturesBatch) =
-            IStakingModule(stakingModule.stakingModuleAddress)
-                .obtainDepositData(_depositsCount, _depositCalldata);
+            if (StakingModuleStatus(stakingModule.status) != StakingModuleStatus.Active) {
```

Comment:
(not so minor as above) not sure whether it's needed to be inside `if` from the prospects of tooling's zero-deposits usage?

Reply:
@avsetsin what will be the desired behavior in this case? In the current implementation, this method doesn't revert when a zero-value deposit is made even in case of a not active module or empty withdrawal credentials.

Reply:
> @avsetsin what will be the desired behavior in this case? In the current implementation, this method doesn't revert when a zero-value deposit is made even in case of a not active module or empty withdrawal credentials.

I believe we should not allow even zero-value deposits if we disabled a module or not set up withdrawal credentials properly